File: Easy/LC226/src/LC226.java
/**
 * Leetcode #226 Problem Link -
 * https://leetcode.com/problems/invert-binary-tree/
 * 
 * @author ketanmalik June 1 2020
 * 
 */

public class LC226 {
	class Solution {
		public TreeNode invertTree(TreeNode root) {
			invertTreeRec(root);
			return root;
		}

		public  void invertTreeRec(TreeNode root) {
			if (root == null)
				return;
			invertTreeRec(root.left);
			invertTreeRec(root.right);
			TreeNode temp = root.left;
			root.left = root.right;
			root.right = temp;
		}
	}
}

File: Easy/LC226/LC226.js
/**
 * Leetcode #226 Problem Link - https://leetcode.com/problems/invert-binary-tree/
 *
 * @author ketanmalik
 * June 1 2020
 *
 */

var invertTree = function (root) {
  invertTreeRec(root);
  return root;
};

var invertTreeRec = function (root) {
  if (root == null) return;
  invertTreeRec(root.left);
  invertTreeRec(root.right);
  let temp = root.left;
  root.left = root.right;
  root.right = temp;
};
